test/ruby/test_process.rb

```diff
@@ -2684,7 +2684,7 @@ def test_concurrent_group_and_pid_wait
       short_wpipe.close # Make short_pid exit
 
       # The short pid has exited, so -1 should pick that up.
-      assert_equal short_pid, Process.waitpid(-1)
+      assert_equal short_pid, Process.waitpid(-1, waitpid_flags)
```

Comment:
This might be a little racy since `short_pid` may not have exited by the time `waitpid` is called.